src/libraries/System.Text.RegularExpressions/gen/DiagnosticDescriptors.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace System.Text.RegularExpressions.Generator
+{
+    internal static class DiagnosticDescriptors
+    {
+        public static DiagnosticDescriptor InvalidRegexGeneratorAttribute { get; } = new DiagnosticDescriptor(
+            id: "SYSLIB1040",
+            title: new LocalizableResourceString(nameof(SR.InvalidRegexGeneratorAttributeTitle), SR.ResourceManager, typeof(FxResources.System.Text.RegularExpressions.Generator.SR)),
+            messageFormat: new LocalizableResourceString(nameof(SR.InvalidRegexGeneratorAttributeMessage), SR.ResourceManager, typeof(FxResources.System.Text.RegularExpressions.Generator.SR)),
+            category: "RegexGenerator",
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            customTags: WellKnownDiagnosticTags.NotConfigurable);
```

Comment:
Will these be documented somewhere? Might be good to attach help links (in a follow-up PR, after they're documented)

Reply:
Eventually